### ISSUE TITLE
Need a way to manually config IPFS to run in 'browser mode'.

### ISSUE THREAD

**Issue by linonetwo** [ipfs/js-ipfs#1259]
Version: 0.28.1
Platform: Mac

#### Description:
The code:
```js
const repoPath = `ipfs-${Math.random()}`;
this.node = new IPFS({ repo: repoPath });
```
is ambiguous.

In [browser environment](https://github.com/linonetwo/ipfs-uploader-browser/blob/master/src/FileUpload.js#L52) it will create an in memory repo (I guess, since nothing created in the fs), but in [electron renderer process](https://github.com/linonetwo/pants-control/blob/master/src/ipfs/ipfs.js#L12) it will create a repo folder in the fs:
<img width="287" alt="wx20180311-184531 2x" src="https://user-images.githubusercontent.com/3746270/37252506-d129135e-255c-11e8-82eb-64dabdecefda.png">

And in electron renderer process I can't simply ```import IPFS from 'ipfs'```, since this will cause ```Can't resolve './build/Release/fs-ext'```.
I can only ```const IPFS = require('electron').remote.require('ipfs');```, and this will create a repo in the fs.
I think the reason is that IPFS is now in 'nodeJS mode', not the 'browser mode' I need.

I wonder if there is a way to manually config IPFS to run in 'browser mode'.

#### Steps to reproduce the error:
clone https://github.com/linonetwo/pants-control , run ```yarn``` and ```yarn start```.

**Comment by diasdavid** [comment]
js-ipfs will use IndexedDB in the browser. You need to "bundle" your code before using the browser process or just use the dist version of js-ipfs. See also https://github.com/ipfs/notes/issues/256

**Comment by linonetwo** [comment]
Actually, I tried, there is something wrong with renderer process:
```html
  <head>
    <meta charset="utf-8">
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:400,400i,700,700i&subset=latin-ext" >
    <script src="https://unpkg.com/ipfs/dist/index.js"></script>

    <title>Pants-Control</title>
  </head>
```
This loads it:
![2018-03-12 10 13 14](https://user-images.githubusercontent.com/3746270/37262430-0e11dcd4-25de-11e8-898f-eced97995f92.png)
But still:
![2018-03-12 10 13 54](https://user-images.githubusercontent.com/3746270/37262443-1d6dfd20-25de-11e8-8746-a2de637c123e.png)

**Comment by linonetwo** [comment]
> You need to "bundle" your code before using the browser process
I'm using create-react-app, and load localhost:3000 into electron's renderer process. I think create-react-app has bundle my code.

**Comment by linonetwo** [comment]
Can you release a ```ipfs-web``` or ```ipfs-browser``` to the npm?
https://github.com/ipfs/js-ipfs#use-in-the-browser has difficulty to use in the renderer process.

**Comment by linonetwo** [comment]
Thank you! I was too careless to found this case sensitive variable. Now application runs both on the renderer process and browser.

And thanks for your advice. I actually bundled my code and it's an independent web page, but you remind me that in my npm script there is a ```cross-env BROWSER=none```, so when I bundled my web page for the renderer process, ```browser``` field in js-ipfs' package.json will not be triggered.

Problem solved, closing this!
